### PR TITLE
EVG-3087 Convert existing Github PR Status API code to use event driven notifications

### PR DESCRIPTION
**Pull request by richardsamuels**
The remaining code in units/github_status_api can't be fully integrated into event driven notifications.

Bad config -> no patch is created.

For Auth request and patch pending, putting these into Event driven notifications risks out of order notifications, and users will not receive any immediate feedback that Evergreen has been notified about the PR.

### CODE REVIEW

File: model/event/subscriptions.go [hunk 1]
```diff
@@ -339,3 +339,17 @@ func NewPatchOutcomeSubscriptionByOwner(owner string, sub Subscriber) Subscripti
 		Subscriber: sub,
 	}
 }
+
+func NewBuildOutcomeSubscriptionByVersion(versionID string, sub Subscriber) Subscription {
+	return Subscription{
+		Type:    ResourceTypeBuild,
+		Trigger: "outcome",
```
Comment (bsamek) [inline model/event/subscriptions.go]: Should "outcome" be a constant?

File: units/github_status_api.go [hunk 2]
```diff
@@ -138,211 +93,122 @@ func NewGithubStatusUpdateJobForBadConfig(intentID string) amboy.Job {
...
 
-		default:
-			return errors.New("build status is pending; refusing to update status")
-		}
+	} else if j.UpdateType == githubUpdateTypeRequestAuth {
+		status.URL = fmt.Sprintf("%s/patch/%s", j.urlBase, j.FetchID)
+		status.Context = "evergreen"
+		status.Description = "patch must be manually authorized"
+		status.State = message.GithubStateFailure
 	}
```
Comment (bsamek) [inline units/github_status_api.go]: You removed the default case, so this no longer returns early.

File: units/github_status_api.go [hunk 3]
```diff
@@ -138,211 +93,122 @@ func NewGithubStatusUpdateJobForBadConfig(intentID string) amboy.Job {
...
-			return err
+			return nil, err
 		}
 		if patchDoc == nil {
-			return errors.New("can't find patch")
+			return nil, errors.New("can't find patch")
 		}
 	}
 
-	if j.UpdateType == githubUpdateTypePatchWithVersion {
```
Comment (bsamek) [inline units/github_status_api.go]: Where was this moved to?

File: units/github_status_api.go (hunk 2 above)
Comment (richardsamuels) [reply units/github_status_api.go]: As discussed in person, this is fine

File: units/github_status_api.go (hunk 3 above)
Comment (richardsamuels) [reply units/github_status_api.go]: As discussed in person, this is fine

**Pull request merged**